inst/include/Rcpp/sugar/functions/var.h

```diff
@@ -34,10 +34,10 @@ class Var : public Lazy< double , Var<RTYPE,NA,T> > {
     Var( const VEC_TYPE& object_ ) : object(object_){}
 
     double get() const{
-        double
-            m = mean(object).get(),
-            ssq = sum( pow(object, 2.0) ).get();
-        return (ssq - m * m * object.size()) / (object.size() - 1);
+        const double
+            average = mean(object).get(),
+            sum_squared_deviations = sum( pow(object - average, 2.0) ).get();
+        return sum_squared_deviations / (object.size() - 1);
     }
```

Comment:
Found the reason.

For `IntegerVector` temporary `object - average` has type `Rcpp::sugar::Minus_Vector_Primitive<13, true, Rcpp::Vector<13, Rcpp::PreserveStorage>>.

In contrast, for `NumericVector` temporary `object - average` has type `Rcpp::sugar::Minus_Vector_Primitive<14, true, Rcpp::Vector<14, Rcpp::PreserveStorage>>`.

This results in invalid deviations calculation (thus resulting in invalid `sum_squared_deviations`).

Rewriting.